File: 11-container-with-most-water/11-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left, right = 0, len(height) - 1
        total = 0
        while(left < right):
            min_height = min(height[left], height[right])
            total = max((right - left )*min_height,total)
            if min_height == height[left]:
                left = left + 1
            else:
                right = right - 1
        return total
        

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int N = height.length;
        int left = 0, right = N - 1;
        
        
        
        int res = 0;
        while(left < right){
            res = Math.max(res,Math.min(height[left], height[right]) * (right - left));
            int cur_min = Math.min(height[left], height[right]);
            if(cur_min == height[left]){
                int temp = left;
                while(temp < right && height[temp] <= height[left]){
                    temp++;
                }
                left = temp;
            }else if (cur_min == height[right]){
                int temp = right;
                while(temp > left && height[temp] <= height[right]){
                    temp--;
                }
                right = temp;                
            }
        }
        return res;
        
        
    }
}